src/ray/gcs/client.cc

```diff
@@ -36,10 +99,36 @@ AsyncGcsClient::AsyncGcsClient() : AsyncGcsClient(ClientID::from_random()) {}
 
 Status AsyncGcsClient::Connect(const std::string &address, int port) {
   RAY_RETURN_NOT_OK(context_->Connect(address, port));
+  // Now initialize shards
+  redisContext *redisContext = context_->sync_context();
+  std::vector<std::string> db_shards_addresses;
+  std::vector<int> db_shards_ports;
+  getRedisShards(redisContext, db_shards_addresses, db_shards_ports);
+
+
+  all_contexts_.push_back(context_);  // Sharding
```

Comment:
I kind of leave out the "use_task_shards" in @stephanie-wang 's original commit. 
What does that part do?